internal/rsl/rsl.go

```diff
@@ -75,6 +82,18 @@ func (e *ReferenceEntry) GetID() gitinterface.Hash {
 
 // Commit creates a commit object in the RSL for the ReferenceEntry.
 func (e *ReferenceEntry) Commit(repo *gitinterface.Repository, sign bool) error {
+	latestEntry, err := GetLatestEntry(repo)
+	if err == nil {
+		e.Number = latestEntry.GetNumber() + 1
+	} else {
+		if errors.Is(err, ErrRSLEntryNotFound) {
+			// First entry
+			e.Number = 1
```

Comment:
We should add docs for this, particularly if we're not starting index 0

Reply:
Added to docstrings